src/RecordSession.cc

```diff
@@ -95,12 +153,19 @@ static int get_num_cpus() {
  * Pick a CPU at random to bind to, unless --cpu-unbound has been given,
  * in which case we return -1.
  */
-static int choose_cpu(RecordSession::BindCPU bind_cpu) {
+static int choose_cpu(int bind_cpu) {
   if (bind_cpu == RecordSession::UNBOUND_CPU) {
     return -1;
   }
+  else if (bind_cpu >= 0) {
```

Comment:
Just drop the `else`. Also, move the comment explaining why we're binding to a CPU up to here.